File: src/main/java/models/Supply.java

package models;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import models.Position;
import models.Supplier;

public class Supply {

    @SerializedName("id")
    @Expose(serialize = false)
    private int id;
    @SerializedName("user")
    @Expose
    private String user;
    @SerializedName("date")
    @Expose
    private String date;
    @SerializedName("act_num")
    @Expose
    private String actNum;
    @SerializedName("supplier")
    @Expose
    private Supplier supplier;
    @SerializedName("positions")
    @Expose
    private List<Position> positions = null;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getActNum() {
        return actNum;
    }

    public void setActNum(String actNum) {
        this.actNum = actNum;
    }

    public Supplier getSupplier() {
        return supplier;
    }

    public void setSupplier(Supplier supplier) {
        this.supplier = supplier;
    }

    public List<Position> getPositions() {
        return positions;
    }

    public void setPositions(List<Position> positions) {
        this.positions = positions;
    }

}

File: src/main/kotlin/models/Supply.kt
package models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import models.position.PositionSupplyFull

class Supply(@SerializedName("id")
             @Expose(serialize = false)
             var id: Int,

             @SerializedName("user")
             @Expose
             var user: String,

             @SerializedName("date")
             @Expose
             var date: String,

             @SerializedName("document")
             @Expose
             var document: String,

             @SerializedName("document_info")
             @Expose
             var documentInfo: String,

             @SerializedName("document_date")
             @Expose
             var documentDate: String,

             @SerializedName("supplier")
             @Expose
             var supplier: Supplier,

             @SerializedName("positions")
             @Expose
             var positions: List<PositionSupplyFull>)
